### PR TITLE
Waiting Room feature for private rooms.

### PR DESCRIPTION
**Pull request by Hasintha-Jayasinghe**
waiting room

### CODE REVIEW

**Comment by Hasintha-Jayasinghe** [comment]
I think the checks failed because of missing environment variables. It should work once you assign some

**Review by MadanBhandari — APPROVED** [review]

File: kofta/src/app/components/InvitedToJoinRoomModal.tsx
Comment (Yawanseh) [inline kofta/src/app/components/InvitedToJoinRoomModal.tsx]: nit: Nested ternaries are not a good practice as it is hard to read and looks a bit confusing

**Review by Yawanseh — COMMENTED** [review]
@benawad I'd prefer if we add [codeclimate](https://codeclimate.com/) to control complexity and duplicates

File: kofta/src/app/components/InvitedToJoinRoomModal.tsx
Comment (Hasintha-Jayasinghe) [reply kofta/src/app/components/InvitedToJoinRoomModal.tsx]: I understand, I'll replace it with a proper switch or if statement later. I just wanted to push this feature as quickly as I could

Comment (HarrisonMayotte) [reply kofta/src/app/components/InvitedToJoinRoomModal.tsx]: I'd recommend just making the change while we await review from Ben.

File: kofta/src/app/useMainWsHandler.ts
Comment (benawad) [inline kofta/src/app/useMainWsHandler.ts:421]: what do you mean it doesn't work?

File: kousa/lib/kousa/room.ex
Comment (benawad) [inline kousa/lib/kousa/room.ex:352]: why did you do this?

File: kofta/src/app/useMainWsHandler.ts
Comment (benawad) [inline kofta/src/app/useMainWsHandler.ts:421]: what do you mean it doesn't work?

File: kousa/lib/kousa/room.ex
Comment (benawad) [inline kousa/lib/kousa/room.ex:290]: why are you doing 2 casts here

File: kousa/lib/kousa/room.ex
Comment (Hasintha-Jayasinghe) [reply kousa/lib/kousa/room.ex:352]: To exit the function. Otherwise the user will automatically join the room

File: kousa/lib/kousa/room.ex
Comment (Hasintha-Jayasinghe) [reply kousa/lib/kousa/room.ex:290]: I send one to the host of the room to show the modal, and the other one goes to the user.

File: kofta/src/app/useMainWsHandler.ts
Comment (Hasintha-Jayasinghe) [reply kofta/src/app/useMainWsHandler.ts:421]: It doesn't trigger the function when the backend sends that message. I checked if the backend is sending the cast by logging to the console, that showed that it was sending the message which means somewhere in the frontend it's not receiving the message

**Comment by Hasintha-Jayasinghe** [comment]
So... is this getting merged or..

**Comment by HarrisonMayotte** [comment]
Pending a second review from Ben, it may be merged, but all feature PRs are on hold until the new design is finished.
Edit: Maybe not a second "review" but resolution of his comments on the main review.

**Review by jabezborja — APPROVED** [review]
H

**Comment by TheOtterlord** [comment]
@Hasintha-Jayasinghe fix conflicts please 🙂

**Comment by Hasintha-Jayasinghe** [comment]
Gimme 30 minutes....

**Comment by Hasintha-Jayasinghe** [comment]
image
What's up with this?
@TheOtterlord

**Comment by TheOtterlord** [comment]
> 
> 
> image
> What's up with this?
> @TheOtterlord

I'm afraid I can't help much here, I have no clue, but if you want, you could head in to the [discord](https://discord.com/invite/wCbKBZF9cV) and try asking one of the `ReactDoge` there? (edit: still learning typescript)

**Comment by Hasintha-Jayasinghe** [comment]
I did that. Waiting for a response.

**Comment by overlisted** [comment]
why did you also reformat random stuff

**Comment by Hasintha-Jayasinghe** [comment]
Because git wouldn't let me commit otherwise. It said something like "ipcRenderer is not redefined change it from let to const"

**Comment by overlisted** [comment]
you could skip git hooks but ok

**Comment by Hasintha-Jayasinghe** [comment]
Hehe I didn't know that!

**Comment by benawad** [comment]
I will port this over to kibbeh later

**Pull request closed without merging**
